notation.go

```diff
@@ -156,6 +160,9 @@ type VerifyOptions struct {
 
 	// PluginConfig is a map of plugin configs.
 	PluginConfig map[string]string
+
+	// UserMetadata contains key-value pairs that must be present in the signature
+	UserMetadata map[string]string
```

Comment:
[ocispec.Descirptor](https://pkg.go.dev/github.com/opencontainers/image-spec/specs-go/v1#Descriptor) already has an field of `Annotations`. Should we reuse it?

Reply:
i don't think it makes sense for verify since a descriptor is not passed into verify, but will change for sign